### PR TITLE
move Future.copy! to Base (fix #29173)

### LINKED ISSUE #29173: move `Future.copy!` to Base
```
help?> copy!
search: copy! copyto! circcopy! unsafe_copyto! copy copysign deepcopy

  copy!(dst, src)

  In-place copy of src into dst. After the call to copy!, dst must be left
  equal to src, otherwise an error is thrown; this function appropriately
  resizes dst if necessary. See also copyto!.
```
Well, sounds convenient for a preallocated `dst::Vector{T}, src::AbstractVector{T}`. Alas,
```
julia> methods(copy!)
# 4 methods for generic function "copy!":
[1] copy!(dst::Random.MersenneTwister, src::Random.MersenneTwister) in Random at /build/julia/src/julia/usr/share/julia/stdlib/v1.0/Random/src/RNGs.jl:142
[2] copy!(dst::Random.DSFMT.DSFMT_state, src::Random.DSFMT.DSFMT_state) in Random.DSFMT at /build/julia/src/julia/usr/share/julia/stdlib/v1.0/Random/src/DSFMT.jl:37
[3] copy!(a::LibGit2.GitCredential, b::LibGit2.GitCredential) in LibGit2 at /build/julia/src/julia/usr/share/julia/stdlib/v1.0/LibGit2/src/gitcredential.jl:83
[4] copy!(dest::BitSet, src::BitSet) in Base at bitset.jl:58
```
I guess the docs went out of sync and `copy!` was deprecated for `( length(dst)==length(src) || resize!(dst, length(src)) ); copyto!(dst, src)` at some point? I'm not sure whether the documented functionality really needs to exist, but either way, reality and doc should match.

For the next release, if `copy!` does not come back, can we rename the other `copy!` methods to something less prominent then? The `BitSet!`  `copy!` probably needs to stay for now, since it is in base and would be breaking?

### PR DESCRIPTION
**Pull request by rfourquet**
As was already discussed, this version is still a bit restrictive, in that we somehow require that the containers be equal after the call to `copy!`. 
For example, should we call the following function `copy!` or `copyto!` ? `copy_or_copyto!(s::Set, a::AbstractArray) = union!(empty!(s), a)` ? This can also be discussed in the ~future~`Future`.

### CODE REVIEW

**Comment by fredrikekre** [comment]
Include a deprecation for `Future.copy!`?

**Comment by rfourquet** [comment]
> Include a deprecation for Future.copy!?

Good point! Or maybe even keep it in there (without deprecation, as 1.x is meant to be stable) until 2.0 ?

**Comment by fredrikekre** [comment]
> > Include a deprecation for Future.copy!?
> 
> Good point! Or maybe even keep it in there (without deprecation, as 1.x is meant to be stable) until 2.0 ?

Yea, I don't know what we have decided with here. There was talk about disabling deprecation warnings durin 1.X so that we can add deprecations but I don't know where we ended up. See also https://github.com/JuliaLang/Juleps/issues/51

**Comment by StefanKarpinski** [comment]
Eh, just leave it there with a comment for now. I for one don't have the energy for any more deprecations at this point.

File: stdlib/Future/src/Future.jl [hunk 1]
```diff
@@ -8,6 +8,9 @@ using Random
 
 ## copy!
 
+# This has now been moved to Base (#29178), and should be deprecated in the
+# next "deprecation phase".
+
```
Comment (KristofferC) [inline stdlib/Future/src/Future.jl]: Cant these just call the Base methods now?

**Review by JeffBezanson — APPROVED** [review]

**Pull request merged**